<launch>
  <arg name="name" default="concat_six" />
  <arg name="cloud_in1" default="front/velodyne" />
  <arg name="cloud_in2" default="rear/velodyne" />
  <arg name="cloud_in3" default="front/left/safety" />
  <arg name="cloud_in4" default="front/right/safety" />
  <arg name="cloud_in5" default="rear/left/safety" />
  <arg name="cloud_in6" default="rear/right/safety" />
  <arg name="cloud_out" default="cloud_merged" />
  <arg name="target_frame" default="vehicle_1/base_link" />
  <arg name="nodelet_manager" default="nodelet_manager" />

  <node pkg="nodelet" type="nodelet"
    args="load lidar/ConcatenateSixClouds $(arg nodelet_manager)"
    name="$(arg name)" output="screen">
    <remap from="cloud_in1" to="$(arg cloud_in1)" />
    <remap from="cloud_in2" to="$(arg cloud_in2)" />
    <remap from="cloud_in3" to="$(arg cloud_in3)" />
    <remap from="cloud_in4" to="$(arg cloud_in4)" />
    <remap from="cloud_in5" to="$(arg cloud_in5)" />
    <remap from="cloud_in6" to="$(arg cloud_in6)" />
    <remap from="cloud_out" to="$(arg cloud_out)" />
    <param name="target_frame" value="$(arg target_frame)" />
  </node>
</launch>